dvc/repo/run.py

```diff
@@ -20,27 +18,10 @@ def run(
 ):
     from dvc.stage.utils import check_graphs, create_stage_from_cli
 
-    if not kwargs.get("cmd"):
-        raise InvalidArgumentError("command is not specified")
-
-    stage_name = kwargs.get("name")
-    if stage_name and single_stage:
-        raise InvalidArgumentError(
-            "`-n|--name` is incompatible with `--single-stage`"
-        )
-
-    if stage_name and fname:
-        raise InvalidArgumentError(
-            "`--file` is currently incompatible with `-n|--name` "
-            "and requires `--single-stage`"
-        )
-
-    if not stage_name and not single_stage:
-        raise InvalidArgumentError("`-n|--name` is required")
-
     stage = create_stage_from_cli(
         self, single_stage=single_stage, fname=fname, **kwargs
     )
+
     if kwargs.get("run_cache", True) and stage.can_be_skipped:
         return None
```

Comment:
@efiop, let me know if we want to remove this behaviour of `Stage is cached, skipping` in 2.0. If so, I'll get rid of it in the next PR.

Reply:
@skshetry Yeah, could totally drop it or refactor it. The functionality is partially covered by run-cache, but not fully. But this `can_be_skipped` is flawed by itself :(